doc/src/modules/vector/api/classes.rst

```diff
@@ -2,13 +2,13 @@
 Essential Classes in sympy.vector (doctrings)
 =============================================
 
-CoordSysCartesian
-=================
+CoordSystem3D
```

Comment:
> The issue with the one class is that every time a new coordinate system is initialized is that the user would have to define the differential_class which could become cumbersome.

Users could use shortcuts or factory functions, or special abbreviations to be passed to the constructor. In SymPy, constructor parameters have to be intended as the instance defining properties.

For example, you may add a parser transforming `differential_class="spherical"` to the proper _Lambda_ expression.

> the methods in a single class could bloat up pretty easily, making it difficult to follow.

The differential operators need not be defined separately for every coordinate system, they are generally defined in terms of the metric of the coordinate system.

When you subclass, you are supposed to define new methods and/or fields in the subclass. What are the new methods you are defining for spherical coordinates wrt Cartesian? None, they are the same. Of course, if someone doesn't know how to write code for differential operators general to all coordinate systems, he may end up rewriting the code in specific methods for every coordinate system. I started my PR in order to avoid this possibility to happen in the future, and only use a single class, as that's the correct way to proceed and avoid messing up SymPy.

The only point in favor of creating a subclass is the special case of 3 dimensions, where you can define the _curl_ which is not generalizable to _N_ dimensions (or at least, not easily/uniquely generalizable). But even in this case, it would be easier to use the same class and just raise an exception if _curl_ is invoked in _N != 3_ dimensions.